console/src/test/scala/io/shiftleft/console/wsserver/WebSocketServerTests.scala

```diff
@@ -0,0 +1,78 @@
+package io.shiftleft.console.wsserver
+
+import java.net.URLEncoder
+import java.util.UUID
+
+import org.scalatest.{Matchers, WordSpec}
+
+import scala.concurrent._
+import duration.Duration.Inf
+import castor.Context.Simple.global
+import cask.util.Logger.Console._
+import io.shiftleft.console.embammonite.EmbeddedAmmonite
+
+import scala.util.Try
+
+class WebSocketServerTests extends WordSpec with Matchers {
+  "WebsocketServer" should {
+
+    "allow connecting, posting a query, and fetching result" in Fixture() { host =>
+      var wsPromise = scala.concurrent.Promise[String]
+      cask.util.WsClient.connect(s"$host/connect") {
+        case cask.Ws.Text(msg) => wsPromise.success(msg)
+      }
+
+      // Connect to websocket endpoint
+
+      val wsMsg = Await.result(wsPromise.future, Inf)
```

Comment:
Not sure how the test will behave if the implementation blocks for too long, I'd add a timeout

```suggestion
      val wsMsg = Await.result(wsPromise.future, Duration(10, SECONDS))
```